decidim-core/lib/decidim/has_reference.rb

```diff
@@ -28,12 +28,17 @@ def calculate_reference
         Decidim.resource_reference_generator.call(self, feature)
       end
 
-      # Internal: Sets the unique reference to the model.
+      # Internal: Sets the unique reference to the model. Note that if the resource
+      # implements `Decidim::Traceable` this method will not create a version.
       #
       # Returns nothing.
       def store_reference
         self[:reference] ||= calculate_reference
-        save if changed?
+        return unless changed?
+
+        # rubocop:disable Rails/SkipsModelValidations
```

Comment:
😔

Reply:
Is the documentation clear enough as to why we need to do this? Maybe I should improve it :(

Reply:
I've improved the documentation to note why we need this.

Reply:
It's an ugly hack, but I can't see any ways to work around it. It's PostgreSQL/Rails/MVC's fault I believe...